File: Baekjoon/17000~17999/[17212]달나라 토끼를 위한 구매대금 지불 도우미/solution.py
N = int(input())
MAX = 100001
dp = [0 for i in range(MAX)]

dp[1] = dp[2] = dp[5] = dp[7] = 1
dp[3] = dp[4] = dp[6] = 2

for i in range(8, N + 1):
    dp[i] = min({dp[i-1] + 1, dp[i-2] + 1, dp[i-5] + 1, dp[i-7] + 1})

print(dp[N])

File: Baekjoon/17000~17999/[17212]달나라 토끼를 위한 구매대금 지불 도우미/solution.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

#define endl "\n"
#define MAX 100'001

vector<int> dp;
int N;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  cin >> N;

  dp = vector<int>(MAX, 0);

  dp[1] = dp[2] = dp[5] = dp[7] = 1;
  dp[3] = dp[4] = dp[6] = 2;

  for (int i = 8; i <= N; i++)
    dp[i] = min({dp[i - 1] + 1, dp[i - 2] + 1, dp[i - 5] + 1, dp[i - 7] + 1});

  cout << dp[N] << endl;

  return 0;
}